include/exadg/operators/operator_base.cpp

```diff
@@ -2153,6 +2155,173 @@ OperatorBase<dim, Number, n_components>::evaluate_face_integrals() const
+
+  unsigned int const n_cells = matrix_free->n_cell_batches() * vectorization_length;
+
+  // assemble a temporary sparse matrix to cut out the blocks
+  SparseMatrix                   tmp_matrix;
+  dealii::DynamicSparsityPattern dsp;
+  internal_init_system_matrix(tmp_matrix, dsp, dof_handler.get_communicator());
+  internal_calculate_system_matrix(tmp_matrix);
+
+  // cut out overlapped block matrices
+  std::vector<dealii::FullMatrix<Number>> overlapped_block_matrix(
+    n_cells, dealii::FullMatrix<Number>(dofs_per_cell));
+
+  std::vector<std::vector<dealii::types::global_dof_index>> all_dof_indices(
+    n_cells, std::vector<dealii::types::global_dof_index>(dofs_per_cell));
+
+  initialize_dof_vector(weights);
+  for(unsigned int cell_batch_index = 0; cell_batch_index < matrix_free->n_cell_batches();
+      ++cell_batch_index)
+  {
+    unsigned int const n_filled_lanes =
+      matrix_free->n_active_entries_per_cell_batch(cell_batch_index);
+
+    for(unsigned int v = 0; v < n_filled_lanes; ++v)
+    {
+      auto const & cell = matrix_free->get_cell_iterator(cell_batch_index, v);
+
+      auto & local_dof_indices = all_dof_indices[cell_batch_index * vectorization_length + v];
+      if(is_mg)
+        cell->get_mg_dof_indices(local_dof_indices);
+      else
+        cell->get_dof_indices(local_dof_indices);
+
+      for(auto const & i : local_dof_indices)
+        weights[i] += 1.;
+    }
+  }
+  weights.compress(dealii::VectorOperation::add);
+
+  for(unsigned int i = 0; i < weights.size(); ++i)
+  {
+    if(weights.in_local_range(i))
+      weights[i] = 1. / std::sqrt(weights[i]);
+  }
+  weights.update_ghost_values();
+
+  dealii::SparseMatrixTools::restrict_to_full_matrices<SparseMatrix,
+                                                       dealii::DynamicSparsityPattern,
+                                                       Number>(tmp_matrix,
+                                                               dsp,
+                                                               all_dof_indices,
+                                                               overlapped_block_matrix);
+
+  // factorize and store cell matrices
+  matrices.resize(n_cells, CellMatrix(dofs_per_cell));
+  for(unsigned int cell_batch_index = 0; cell_batch_index < matrix_free->n_cell_batches();
+      ++cell_batch_index)
+  {
+    unsigned int const n_filled_lanes =
+      matrix_free->n_active_entries_per_cell_batch(cell_batch_index);
+
+    for(unsigned int v = 0; v < n_filled_lanes; ++v)
+    {
+      // get overlapped cell matrix
+      auto const & overlapped_cell_matrix =
+        overlapped_block_matrix[cell_batch_index * vectorization_length + v];
+
+      // store the cell matrix and renumber lexicographic if necessary
+      auto & lapack_matrix = matrices[cell_batch_index * vectorization_length + v];
+      if(is_dg)
+      {
+        lapack_matrix = overlapped_cell_matrix;
+      }
```

Comment:
This is similar as the other comment on `is_dg`. It looks like the whole system matrix is assembled in the DG case? Does weights = 1 hold in the DG case?

Reply:
For this special case of elementwise additive Schwarz and DG yes, we have only one contribution per dof, meaning weight = 1. For this special case additive Schwarz and Block Jacobi are identical. One can interpret Block Jacobi as a special additive Schwarz case.

Reply:
From a software side, it looks like we should make additive Schwarz and block-Jacobi "disjunct" implementations?

Reply:
Yes, I think so. The only option I see is if we detect block-Jacobi behaviour of additive Schwarz we use specialised code paths for performance optimization (like we do here in the PR for the application of the element wise inverse).